pkg/apis/experimental/types.go

```diff
@@ -461,3 +461,75 @@ type JobCondition struct {
 	// Human readable message indicating details about last transition.
 	Message string `json:"message,omitempty"`
 }
+
+// IngressPoint encapsulates the inputs needed to connect an alias to endpoints.
+// It represents a rule collection of inbound connections from the external network
+// that would be satisfied by a load balancer.
+type IngressPoint struct {
+	api.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+	api.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the desired state of the IngressPoint.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+	Spec IngressPointSpec `json:"spec,omitempty"`
+
+	// Status is the current state of the IngressPoint.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+	Status IngressPointStatus `json:"status,omitempty"`
+}
+
+// IngressPointList is a collection of IngressPoints,
+type IngressPointList struct {
+	api.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+	api.ListMeta `json:"metadata,omitempty"`
+
+	// Items is the list of ingressPoints.
+	Items []IngressPoint `json:"items"`
+}
+
+// ServiceRef represents a reference to a single service:port.
+type ServiceRef struct {
+	// Define the name of the referenced service.
+	Name string `json:"name,omitempty"`
+
+	// Define the namespace of the referenced service.
+	Namespace string `json:"namespace,omitempty"`
+
+	// Define the service's port which the http traffic will be forwarded to.
+	Port int `json:"port,omitempty"`
+}
+
+// PathRef represents a single load balancer rule of forwarding the specified url to the related backend service.
+type PathRef struct {
+	// Domain describe http url's domain.
+	// Examples: "foo.example.com", "bar.example.com".
+	Domain string `json:"domain,omitempty"`
```

Comment:
I tried to play it fast and loose with url in the proposal and got shot down. Apparently a url is something that has a long list of properties (https://tools.ietf.org/html/rfc3986).

I thought about the long names. At first they felt like java. But I liked that I would never have to add comments to my yaml, and I liked being able to justify them with rules. I'm open to bettern name schemes.

I don't understand you substitution, do you mean change Path.* to Url? I want to be explicit about what is used for matching vs the target of the match, somehow. Otherwise we're getting too close to the gce urlmap confusion.